File: android/src/main/java/com/lunarrangepicker/LunarCalendarHelper.kt
package com.lunarrangepicker

import java.util.Calendar
import java.util.Locale

data class LunarDateResult(
    val day: Int,
    val month: Int,
    val year: Int,
    val isLeap: Boolean,
    val lunarDayName: String
)

object LunarCalendarHelper {
    fun convertSolarToLunar(calendar: Calendar, language: String = "vi"): LunarDateResult {
        // Sử dụng ChineseCalendar của Android ICU hoặc thuật toán chuyển đổi GMT+7
        val uCalendar = android.icu.util.ChineseCalendar()
        uCalendar.timeInMillis = calendar.timeInMillis

        val day = uCalendar.get(android.icu.util.ChineseCalendar.DAY_OF_MONTH)
        val month = uCalendar.get(android.icu.util.ChineseCalendar.MONTH) + 1
        val year = uCalendar.get(android.icu.util.ChineseCalendar.YEAR)
        val isLeap = uCalendar.get(android.icu.util.ChineseCalendar.IS_LEAP_MONTH) == 1

        val dayName = when {
            language == "vi" && day == 1 -> "Mùng 1"
            language == "vi" && day == 15 -> "Rằm"
            day < 10 -> "0$day"
            else -> day.toString()
        }

        return LunarDateResult(day, month, year, isLeap, dayName)
    }
}

File: ios/LunarCalendarHelper.swift
import Foundation

public struct LunarDateResult {
    public let day: Int
    public let month: Int
    public let year: Int
    public let isLeap: Bool
    public let lunarDayName: String
}

public class LunarCalendarHelper {
    /// Chuyển đổi từ NSDate sang Âm Lịch (Sử dụng Calendar Chinese chuẩn của hệ thống)
    public static func convertSolarToLunar(date: Date, language: String = "vi") -> LunarDateResult {
        let chineseCalendar = Calendar(identifier: .chinese)
        let components = chineseCalendar.dateComponents([.year, .month, .day, .isLeapMonth], from: date)
        
        let day = components.day ?? 1
        let month = components.month ?? 1
        let year = components.year ?? 1
        let isLeap = components.isLeapMonth ?? false
        
        // Tạo tên hiển thị (Ví dụ: Mùng 1, Rằm hoặc 01/05)
        var dayName = "\(day)"
        if language == "vi" {
            if day == 1 { dayName = "Mùng 1" }
            else if day == 15 { dayName = "Rằm" }
            else if day < 10 { dayName = "0\(day)" }
        }
        
        return LunarDateResult(
            day: day,
            month: month,
            year: year,
            isLeap: isLeap,
            lunarDayName: dayName
        )
    }
}